craft_parts/sources/local_source.py

```diff
@@ -157,13 +157,24 @@ def update(self):
                 os.path.join(self.part_src_dir, file_path),
             )
 
-        # And remove deleted entries
+    def source_cleanup(self):
+        """Remove pulled entries not present in the actual part source."""
         for entry in self._deleted_entries:
             path = Path(self.part_src_dir, entry)
-            if path.is_dir():
-                shutil.rmtree(str(path))
-            else:
-                path.unlink()
+            _delete_entry(path)
+
+            if self.part_build_dir:
+                path = self.part_build_dir / entry
+                _delete_entry(path)
+
+
+def _delete_entry(path: Path) -> None:
+    if path.is_dir():
+        logger.debug("delete dir: %s", path)
```

Comment:
Better %r, or as we don in charmcraft, %r of str(path), to not be confused by spaces but also don't get the ugly "Pathlib" thing